echopype/tests/consolidate/test_consolidate_integration.py

```diff
@@ -147,36 +151,430 @@ def _build_ds_Sv(channel, range_sample, ping_time, sample_interval):
+    ed["Platform"]["pitch"] = ed["Platform"]["pitch"].fillna(0)
+    ed["Platform"]["roll"] = ed["Platform"]["roll"].fillna(0)
+
+    # Compute `depth` using platform angle values
+    ds_Sv = ep.consolidate.add_depth(ds_Sv, ed, use_platform_angles=True)
+
+    # Check history attribute
+    history_attribute = ds_Sv["depth"].attrs["history"]
+    history_attribute_without_time = history_attribute[33:]
+    assert history_attribute_without_time == (
+        ". depth` calculated using: Sv `echo_range`, Echodata `Platform` Angle Data."
+    )
+
+    # Compute transducer depth
+    echo_range_scaling = ek_use_platform_angles(ed["Platform"], ds_Sv["ping_time"])
+
+    # Iterate through every ping time
+    for ping_time_index in range(len(ds_Sv["ping_time"])):
+        # Grab echo range scaling value
+        echo_range_scaling_value = echo_range_scaling.isel(ping_time=ping_time_index)
+        # Iterate through every channel
+        for channel_index in range(len(ds_Sv["channel"])):
+            # Grab vectors
+            depth_vector = ds_Sv["depth"].isel(
+                channel=channel_index, ping_time=ping_time_index
+            )
+            echo_range_vector = ds_Sv["echo_range"].isel(
+                channel=channel_index, ping_time=ping_time_index
+            )
+            # Check if depth vector is equal to echo range scaling value * echo range
+            assert np.all(
+                np.isclose(
+                    (depth_vector).data,
+                    (echo_range_scaling_value * echo_range_vector).data, 
+                    equal_nan=True
+                )
+            )
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("file, sonar_model, compute_Sv_kwaargs", [
+    (
+        "echopype/test_data/ek60/NBP_B050N-D20180118-T090228.raw",
+        "EK60",
+        {}
+    ),
+    (
+        "echopype/test_data/ek60/ncei-wcsd/Summer2017-D20170620-T021537.raw",
+        "EK60",
+        {}
+    ),
+    (
+        "echopype/test_data/ek80/ncei-wcsd/SH1707/Reduced_D20170826-T205615.raw",
+        "EK80",
+        {"waveform_mode":"BB", "encode_mode":"complex"}
+    ),
+    (
+        "echopype/test_data/ek80/ncei-wcsd/SH2106/EK80/Reduced_Hake-D20210701-T131621.raw",
+        "EK80",
+        {"waveform_mode":"CW", "encode_mode":"power"}
+    )
+])
+def test_add_depth_EK_with_beam_angles(file, sonar_model, compute_Sv_kwaargs):
+    """Test `depth` values when using EK Beam angles to compute it."""
+    # Open EK Raw file and Compute Sv
+    ed = ep.open_raw(file, sonar_model=sonar_model)
+    ds_Sv = ep.calibrate.compute_Sv(ed, **compute_Sv_kwaargs)
+
+    # Replace any Beam Angle NaN values with 0
+    ed["Sonar/Beam_group1"]["beam_direction_x"] = ed["Sonar/Beam_group1"]["beam_direction_x"].fillna(0)
+    ed["Sonar/Beam_group1"]["beam_direction_y"] = ed["Sonar/Beam_group1"]["beam_direction_y"].fillna(0)
+    ed["Sonar/Beam_group1"]["beam_direction_z"] = ed["Sonar/Beam_group1"]["beam_direction_z"].fillna(0)
```

Comment:
I suspect that you use these so that the add_depth computation would go through? Different from the pitch, roll angles though, a beam direction vector (0, 0, 0) does not make sense as a direction vector, right? The length of (0, 0, 0) would be 0, but a direction vector should have a length 1, right?

Reply:
I agree yeah that doesn't make sense. I'll add a test to check that `ek_use_beam_angles` outputs what it should output.